### PR TITLE
Port FileConventions to VS17 and NetCore

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Batch/Support/FileConventions/Src/AzureBatchFileConventions/AzureBatchFileConventions.csproj [hunk 1]
```diff
@@ -1,126 +1,44 @@
...
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <DocumentationFile>bin\Release\Microsoft.Azure.Batch.Conventions.Files.XML</DocumentationFile>
+  <PropertyGroup>
+    <SignAssembly>false</SignAssembly>
```
Comment (matthchr) [inline src/SDKs/Batch/Support/FileConventions/Src/AzureBatchFileConventions/AzureBatchFileConventions.csproj]: @shahabhijeet, I am not sure doing this is okay/allowed, please confirm.

Comment (shahabhijeet) [inline src/SDKs/Batch/Support/FileConventions/Src/AzureBatchFileConventions/AzureBatchFileConventions.csproj]: @matthchr ideally we highly discourage not being part of the common build properties.
But as this project is a special case (from what I got to know from our offline conversation)
By default the two properties (SignAssembly, DelaySign) are false, so if your intent is to document explicitly the values, then leave it, else you don't need these properties set for 'False'

**Review by shahabhijeet — APPROVED** [review]

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
This change will eventually fail a Signed build.
So we will not be able to accept this change in it's current form.

**Comment by matthchr** [comment]
@shahabhijeet I think this is fixed now: http://azuresdkci.cloudapp.net/view/NetCore-SDK/job/NetCore-SDK-Sign/131/

**Pull request merged**